core/common/src/test/scala/org/typelevel/otel4s/TextMapGetterSuite.scala

```diff
@@ -0,0 +1,52 @@
+package org.typelevel.otel4s
+
+import munit.FunSuite
+
+import scala.collection.{immutable, mutable}
+
+class TextMapGetterSuite extends FunSuite {
+  // `TextMapGetter[C]` is not implicitly summoned by this method
+  // so that it tests that instances are available in a non-generic
+  // context.
+  def check[C](tmg: TextMapGetter[C])(carrier: C): Unit = {
+    assertEquals(tmg.keys(carrier).toSeq, Seq("1", "2", "3"))
+    assertEquals(tmg.get(carrier, "1"), Some("one"))
+    assertEquals(tmg.get(carrier, "2"), Some("two"))
+    assertEquals(tmg.get(carrier, "3"), Some("three"))
+    assert(tmg.get(carrier, "0").isEmpty)
+    assert(tmg.get(carrier, "4").isEmpty)
+  }
+
+  test("TextMapGetter[Map[String, String]") {
+    check(implicitly[TextMapGetter[mutable.HashMap[String, String]]])(
+      mutable.HashMap("1" -> "one", "2" -> "two", "3" -> "three")
+    )
+    check(implicitly[TextMapGetter[immutable.TreeMap[String, String]]])(
+      immutable.TreeMap("1" -> "one", "2" -> "two", "3" -> "three")
+    )
+    check(implicitly[TextMapGetter[collection.Map[String, String]]])(
+      collection.Map("1" -> "one", "2" -> "two", "3" -> "three")
+    )
+  }
+
+  test("TextMapGetter[Seq[(String, String)]") {
```

Comment:
is there a specification for what's supposed to happen?

Reply:
otel is silent, because it doesn't talk about array-based carriers.  We could either leave it in the realm of undefined behavior, or document that first one wins.

Reply:
I'm worried about us specifying a behaviour and then otel deciding to specify some other behaviour at a later date and leaving us stuck. we could perhaps say that this implementation *happens* to return the first value, but that said behaviour may change at any point, such as if otel specifies a behaviour?

Reply:
I've added notes to each of the docs - could I get your thoughts on them?

Reply:
I think those look great.